flutter-gui-tests/testSrc/io/flutter/tests/gui/ProjectCreator.kt

```diff
@@ -1,60 +1,131 @@
 package io.flutter.tests.gui
 
+import com.intellij.ide.fileTemplates.impl.UrlUtil
 import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.util.io.StreamUtil
+import com.intellij.platform.templates.github.ZipUtil
 import com.intellij.testGuiFramework.framework.Timeouts
 import com.intellij.testGuiFramework.impl.*
 import com.intellij.testGuiFramework.utils.TestUtilsClass
 import com.intellij.testGuiFramework.utils.TestUtilsClassCompanion
+import com.intellij.util.UriUtil
+import com.intellij.util.io.URLUtil
 import org.fest.swing.exception.WaitTimedOutError
+import java.io.BufferedOutputStream
+import java.io.File
+import java.io.FileOutputStream
+import java.net.URL
 
 val GuiTestCase.ProjectCreator by ProjectCreator
 
 // Inspired by CommunityProjectCreator
 class ProjectCreator(guiTestCase: GuiTestCase) : TestUtilsClass(guiTestCase) {
 
-  companion object : TestUtilsClassCompanion<ProjectCreator>({ it -> ProjectCreator(it) })
+  companion object : TestUtilsClassCompanion<ProjectCreator>({ ProjectCreator(it) })
 
   private val defaultProjectName = "untitled"
-  private val LOG = Logger.getInstance(this.javaClass)
+  private val sampleProjectName = "simple_app"
+  private val log = Logger.getInstance(this.javaClass)
+
+  fun importProject(projectName: String = sampleProjectName): File {
+    val projectDirFile = extractProject(projectName)
+    val projectPath: File = guiTestCase.guiTestRule.importProject(projectDirFile)
+    with(guiTestCase) {
+      openMainInProject()
+      waitForFirstIndexing()
+      ideFrame {
+        editor {
+          try {
+            val note = notificationPanel()
+            // Best practice is to avoid reusing strings defined in bundles.
+            if (note?.getLabelText() == "Pubspec has been edited") {
+              log.info("Get dependencies")
+              note.clickLink("Get dependencies")
+              robot().waitForIdle() // TODO(messick) Check Messages view for completion of 'flutter packages get'
+            }
+          }
+          catch (ex: WaitTimedOutError) {
+            // TODO(messick) Check that pubspec is valid.
```

Comment:
👍

Reply:
Maybe at least a log entry here in the meantime?

Reply:
I probably should have done that. I'm writing on that code now so it will be handled in the next PR.